runelite-client/src/main/java/net/runelite/client/plugins/groundmarkers/GroundMarkerMinimapOverlay.java

```diff
@@ -0,0 +1,123 @@
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package net.runelite.client.plugins.groundmarkers;
+
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.util.Collection;
+import javax.inject.Inject;
+import net.runelite.api.Client;
+import net.runelite.api.Perspective;
+import net.runelite.api.Point;
+import net.runelite.api.coords.LocalPoint;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.OverlayPriority;
+
+public class GroundMarkerMinimapOverlay extends Overlay
+{
+	private static final int MAX_DRAW_DISTANCE = 16;
+	private static final int TILE_WIDTH = 4;
+	private static final int TILE_HEIGHT = 4;
+
+	private final Client client;
+	private final GroundMarkerConfig config;
+	private final GroundMarkerPlugin plugin;
+
+	@Inject
+	private GroundMarkerMinimapOverlay(Client client, GroundMarkerConfig config, GroundMarkerPlugin plugin)
+	{
+		this.client = client;
+		this.config = config;
+		this.plugin = plugin;
+		setPosition(OverlayPosition.DYNAMIC);
+		setPriority(OverlayPriority.LOW);
+		setLayer(OverlayLayer.ABOVE_WIDGETS);
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		if (!config.drawTileOnMinimmap())
+		{
+			return null;
+		}
+
+		final Collection<ColorTileMarker> points = plugin.getPoints();
+		for (final ColorTileMarker point : points)
+		{
+			WorldPoint worldPoint = point.getWorldPoint();
+			if (worldPoint.getPlane() != client.getPlane())
+			{
+				continue;
+			}
+
+			Color tileColor = point.getColor();
+			if (tileColor == null || !config.rememberTileColors())
+			{
+				// If this is an old tile which has no color, or rememberTileColors is off, use marker color
+				tileColor = config.markerColor();
+			}
+
+			drawOnMinimap(graphics, worldPoint, tileColor);
+		}
+
+		return null;
+	}
+
+	private void drawOnMinimap(Graphics2D graphics, WorldPoint point, Color color)
+	{
+		WorldPoint playerLocation = client.getLocalPlayer().getWorldLocation();
+
+		if (point.distanceTo(playerLocation) >= MAX_DRAW_DISTANCE)
+		{
+			return;
+		}
+
+		LocalPoint lp = LocalPoint.fromWorld(client, point);
+		if (lp == null)
+		{
+			return;
+		}
+
+		Point posOnMinimap = Perspective.localToMinimap(client, lp);
+		if (point == null)
+		{
+			return;
+		}
+
+		double unit = Math.PI / 1024.0d;
```

Comment:
Move this drawing of squares on the minimap to `OverlayUtil` so other plugins can reuse it.